test/compiler/EscapeAnalysis/EAUtils.jl

```diff
@@ -86,7 +83,7 @@ struct EscapeCacheInfo
 end
 
 struct EscapeCache
-    cache::IdDict{MethodInstance,EscapeCacheInfo}
+    cache::IdDict{MethodInstance,EscapeCacheInfo} # Should this be CodeInstance to EscapeCacheInfo?
```

Comment:
@aviatesk this is something we should talk about. In my head this should probably be an `IdDict{CodeInstance, EscapeCacheInfo}` since we are attaching information to a particular inference task and we store that information in the `CodeInstance`

Reply:
Yeah, I agree. I would like to work on that refactoring to see what possibilities this change opens up, if you want.

Reply:
That would be great!